grpo-finetuning-qwen3/agent-skill/grpo-finetune/run_pipeline.py

```
@@ -0,0 +1,192 @@
+            checkpoint(f"Dataset ready  ·  {row_count} prompts")
+        else:
+            print("Waiting for dataset", end="", flush=True)
+            while True:
+                ds = fw.datasets.get(dataset_id=dataset_id)
+                if ds.state == "READY":
+                    print()
+                    break
+                print(".", end="", flush=True)
+                time.sleep(3)
+            checkpoint(f"Dataset ready  ·  {row_count} prompts")
+    except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)